pypeit/scripts/ql_keck_deimos.py

```diff
@@ -0,0 +1,233 @@
+    pypeit_files = ps.fitstbl.write_pypeit(output_path=output_path,
+                                          cfg_lines=ps.user_cfg,
+                                          configs='all')
+
+    # Loop on setups, rename + run calibs
+    for pypeit_file, setup, ii in zip(pypeit_files, setups, indx):
+
+        # Rename with _calib
+        calib_pypeit_file = pypeit_file.replace('deimos_{}.pypeit'.format(setup),
+                                                'deimos_calib_{}.pypeit'.format(setup))
+        os.rename(pypeit_file, calib_pypeit_file)
+
+        # Run me via the script
+        redux_path = os.path.dirname(calib_pypeit_file)  # Path to PypeIt file
+        run_pargs = run_pypeit.RunPypeIt.parse_args([calib_pypeit_file,
+                                           '-r={}'.format(redux_path),
+                                           '-c'])
+        run_pypeit.RunPypeIt.main(run_pargs)
+
+
+def get_science_setup(pargs, script_Utils):
+    """
+    Figure out the matching setup for the science file
+
+    Args:
+        pargs (argparse.ArgumentParser):
+            Command line arguments
+        script_Utils (:class:`pypeit.scripts.utilts.Utilities`):
+
+    Returns:
+        tuple: int, :class:`pypeit.pypeitsetup.PypeItSetup`
+
+    """
+    # Check file exists
+    science_file = os.path.join(pargs.full_rawpath, pargs.science)
+    if not os.path.isfile(science_file):
+        msgs.error("Your science filename {} does not exist. Check your path".format(science_file))
+    # Run setup
+    ps_sci, _, indx = script_Utils.run_setup(science_file,
+                                              extension='', 
+                                              no_write_sorted=True)
+    # Check against existing PypeIt files
+    pypeit_files = glob.glob(os.path.join(pargs.redux_path, 'keck_deimos_*', 'keck_deimos_calib_*.pypeit'))
+    mtch = []
+    for pypeit_file in pypeit_files:
+        # Read
+        tmp_scriptU = Utilities('keck_deimos', pypeit_file=pypeit_file)
+        # Check for a match
+        match = True
+        for key in tmp_scriptU.spectrograph.configuration_keys():
+            if ps_sci.fitstbl[key][0] != tmp_scriptU.sdict[key]:
+                match = False
+        if match:
+            mtch.append(pypeit_file)
+    if len(mtch) != 1:
+        msgs.error("Matched to more than one setup.  Inconceivable!")
+
+    return mtch[0], ps_sci
+
+def run_on_science(pargs, script_Utils, calib_pypeit_file, ps_sci):
+    """
+    Process a science frame
+
+    Args:
+        pargs (argparse.ArgumentParser):
+            Command line arguments
+        script_Utils:
+        calib_pypeit_file (str):
+        ps_sci (:class:`pypeit.pypeitsetup.PypeItSetup`):
+    """
+    # Load up and slurp PypeIt file
+    ps = PypeItSetup.from_pypeit_file(calib_pypeit_file)
+    # Parse science file info
+    science_file = os.path.join(pargs.full_rawpath, pargs.science)
+    science_pypeit = calib_pypeit_file.replace('calib', 'science')
```

Comment:
I tried running `pypeit_ql_keck_deimos full_path_to_raw_files --science=DE.20130409.20629.fits --slit_spat=3:763 --redux_path=path_for_calibs` and I noticed that Pypeit reduces all the science frames and not only the one specified. Maybe at this point in the code you need to modify  `ps` to remove all the science frames but the one specified in `--science`?

Reply:
good catch.  The purging has been added..